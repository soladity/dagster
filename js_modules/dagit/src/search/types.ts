export enum SearchResultType {
  Asset,
  Page,
  Pipeline,
  Repository,
  Run,
  Schedule,
  Sensor,
  Solid,
}

export type SearchResult = {
  key: string;
  label: string;
  description: string;
  href: string;
  type: SearchResultType;
  tags?: string;
};
