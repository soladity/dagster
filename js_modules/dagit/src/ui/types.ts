export type Spacing = 0 | 1 | 2 | 4 | 8 | 12 | 16 | 20 | 24 | 32 | 48 | 64;
export type DirectionalSpacing =
  | Spacing
  | {
      top?: Spacing;
      right?: Spacing;
      bottom?: Spacing;
      left?: Spacing;
      vertical?: Spacing;
      horizontal?: Spacing;
    };

export type AlignItems =
  | 'stretch'
  | 'center'
  | 'start'
  | 'end'
  | 'flex-start'
  | 'flex-end'
  | 'baseline';

export type FlexDirection = 'row' | 'row-reverse' | 'column' | 'column-reverse';
export type FlexProperties = {
  alignItems?: AlignItems;
  basis?: string;
  direction?: FlexDirection;
  display?: 'flex' | 'inline-flex';
  grow?: number;
  justifyContent?:
    | 'center'
    | 'start'
    | 'end'
    | 'flex-start'
    | 'flex-end'
    | 'left'
    | 'right'
    | 'space-between'
    | 'space-around'
    | 'space-evenly'
    | 'stretch';
  shrink?: number;
  wrap?: 'nowrap' | 'wrap' | 'wrap-reverse';
};

export type BorderSide = 'top' | 'right' | 'bottom' | 'left' | 'horizontal' | 'vertical' | 'all';
export type BorderWidth = 1 | 2;
export type BorderColor = 'light' | 'medium' | 'dark';
export type BorderSetting = {width: BorderWidth; color: BorderColor; side: BorderSide};
