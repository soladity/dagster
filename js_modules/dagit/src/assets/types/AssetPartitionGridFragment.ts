// @generated
/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL fragment: AssetPartitionGridFragment
// ====================================================

export interface AssetPartitionGridFragment_runOrError_PipelineRunNotFoundError {
  __typename: "PipelineRunNotFoundError" | "PythonError";
}

export interface AssetPartitionGridFragment_runOrError_PipelineRun {
  __typename: "PipelineRun";
  pipelineSnapshotId: string | null;
}

export type AssetPartitionGridFragment_runOrError = AssetPartitionGridFragment_runOrError_PipelineRunNotFoundError | AssetPartitionGridFragment_runOrError_PipelineRun;

export interface AssetPartitionGridFragment {
  __typename: "AssetMaterialization";
  partition: string | null;
  runOrError: AssetPartitionGridFragment_runOrError;
}
