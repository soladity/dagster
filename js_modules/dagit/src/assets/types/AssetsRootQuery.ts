// @generated
/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: AssetsRootQuery
// ====================================================

export interface AssetsRootQuery_assetsOrError_AssetsNotSupportedError {
  __typename: "AssetsNotSupportedError";
  message: string;
}

export interface AssetsRootQuery_assetsOrError_AssetConnection_nodes_key {
  __typename: "AssetKey";
  path: string[];
}

export interface AssetsRootQuery_assetsOrError_AssetConnection_nodes {
  __typename: "Asset";
  key: AssetsRootQuery_assetsOrError_AssetConnection_nodes_key;
}

export interface AssetsRootQuery_assetsOrError_AssetConnection {
  __typename: "AssetConnection";
  nodes: AssetsRootQuery_assetsOrError_AssetConnection_nodes[];
}

export type AssetsRootQuery_assetsOrError = AssetsRootQuery_assetsOrError_AssetsNotSupportedError | AssetsRootQuery_assetsOrError_AssetConnection;

export interface AssetsRootQuery {
  assetsOrError: AssetsRootQuery_assetsOrError;
}
