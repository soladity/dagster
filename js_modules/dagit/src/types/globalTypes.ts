// @generated
/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

//==============================================================
// START Enums and Input Objects
//==============================================================

export enum ComputeIOType {
  STDERR = "STDERR",
  STDOUT = "STDOUT",
}

export enum EvaluationErrorReason {
  FIELDS_NOT_DEFINED = "FIELDS_NOT_DEFINED",
  FIELD_NOT_DEFINED = "FIELD_NOT_DEFINED",
  MISSING_REQUIRED_FIELD = "MISSING_REQUIRED_FIELD",
  MISSING_REQUIRED_FIELDS = "MISSING_REQUIRED_FIELDS",
  RUNTIME_TYPE_MISMATCH = "RUNTIME_TYPE_MISMATCH",
  SELECTOR_FIELD_ERROR = "SELECTOR_FIELD_ERROR",
}

export enum LogLevel {
  CRITICAL = "CRITICAL",
  DEBUG = "DEBUG",
  ERROR = "ERROR",
  INFO = "INFO",
  WARNING = "WARNING",
}

export enum ObjectStoreOperationType {
  CP_OBJECT = "CP_OBJECT",
  GET_OBJECT = "GET_OBJECT",
  RM_OBJECT = "RM_OBJECT",
  SET_OBJECT = "SET_OBJECT",
}

export enum PipelineRunStatus {
  FAILURE = "FAILURE",
  MANAGED = "MANAGED",
  NOT_STARTED = "NOT_STARTED",
  STARTED = "STARTED",
  SUCCESS = "SUCCESS",
}

export enum ScheduleStatus {
  ENDED = "ENDED",
  RUNNING = "RUNNING",
  STOPPED = "STOPPED",
}

export enum StepKind {
  COMPUTE = "COMPUTE",
}

export interface ExecutionMetadata {
  runId?: string | null;
  tags?: ExecutionTag[] | null;
}

export interface ExecutionParams {
  selector: ExecutionSelector;
  environmentConfigData?: any | null;
  mode?: string | null;
  executionMetadata?: ExecutionMetadata | null;
  stepKeys?: string[] | null;
  preset?: string | null;
  retryRunId?: string | null;
}

export interface ExecutionSelector {
  name: string;
  solidSubset?: string[] | null;
}

export interface ExecutionTag {
  key: string;
  value: string;
}

export interface PipelineRunsFilter {
  runId?: string | null;
  pipeline?: string | null;
  tagKey?: string | null;
  tagValue?: string | null;
  status?: PipelineRunStatus | null;
}

//==============================================================
// END Enums and Input Objects
//==============================================================
