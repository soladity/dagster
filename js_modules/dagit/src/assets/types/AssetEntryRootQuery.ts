// @generated
/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AssetKeyInput } from "./../../types/globalTypes";

// ====================================================
// GraphQL query operation: AssetEntryRootQuery
// ====================================================

export interface AssetEntryRootQuery_assetOrError_AssetNotFoundError {
  __typename: "AssetNotFoundError";
}

export interface AssetEntryRootQuery_assetOrError_AssetsNotSupportedError {
  __typename: "AssetsNotSupportedError";
  message: string;
}

export interface AssetEntryRootQuery_assetOrError_Asset_key {
  __typename: "AssetKey";
  path: string[];
}

export interface AssetEntryRootQuery_assetOrError_Asset {
  __typename: "Asset";
  key: AssetEntryRootQuery_assetOrError_Asset_key;
}

export type AssetEntryRootQuery_assetOrError = AssetEntryRootQuery_assetOrError_AssetNotFoundError | AssetEntryRootQuery_assetOrError_AssetsNotSupportedError | AssetEntryRootQuery_assetOrError_Asset;

export interface AssetEntryRootQuery {
  assetOrError: AssetEntryRootQuery_assetOrError;
}

export interface AssetEntryRootQueryVariables {
  assetKey: AssetKeyInput;
}
