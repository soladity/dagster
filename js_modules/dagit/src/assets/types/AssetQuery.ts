// @generated
/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AssetKeyInput, PipelineRunStatus } from "./../../types/globalTypes";

// ====================================================
// GraphQL query operation: AssetQuery
// ====================================================

export interface AssetQuery_assetOrError_AssetNotFoundError {
  __typename: "AssetNotFoundError";
}

export interface AssetQuery_assetOrError_Asset_key {
  __typename: "AssetKey";
  path: string[];
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_runOrError_PipelineRunNotFoundError {
  __typename: "PipelineRunNotFoundError" | "PythonError";
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_runOrError_PipelineRun {
  __typename: "PipelineRun";
  id: string;
  runId: string;
  mode: string;
  status: PipelineRunStatus;
  pipelineName: string;
  pipelineSnapshotId: string | null;
}

export type AssetQuery_assetOrError_Asset_assetMaterializations_runOrError = AssetQuery_assetOrError_Asset_assetMaterializations_runOrError_PipelineRunNotFoundError | AssetQuery_assetOrError_Asset_assetMaterializations_runOrError_PipelineRun;

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_stepStats {
  __typename: "PipelineRunStepStats";
  endTime: number | null;
  startTime: number | null;
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventPathMetadataEntry {
  __typename: "EventPathMetadataEntry";
  label: string;
  description: string | null;
  path: string;
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventJsonMetadataEntry {
  __typename: "EventJsonMetadataEntry";
  label: string;
  description: string | null;
  jsonString: string;
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventUrlMetadataEntry {
  __typename: "EventUrlMetadataEntry";
  label: string;
  description: string | null;
  url: string;
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventTextMetadataEntry {
  __typename: "EventTextMetadataEntry";
  label: string;
  description: string | null;
  text: string;
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventMarkdownMetadataEntry {
  __typename: "EventMarkdownMetadataEntry";
  label: string;
  description: string | null;
  mdStr: string;
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventPythonArtifactMetadataEntry {
  __typename: "EventPythonArtifactMetadataEntry";
  label: string;
  description: string | null;
  module: string;
  name: string;
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventFloatMetadataEntry {
  __typename: "EventFloatMetadataEntry";
  label: string;
  description: string | null;
  floatValue: number;
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventIntMetadataEntry {
  __typename: "EventIntMetadataEntry";
  label: string;
  description: string | null;
  intValue: number;
}

export type AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries = AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventPathMetadataEntry | AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventJsonMetadataEntry | AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventUrlMetadataEntry | AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventTextMetadataEntry | AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventMarkdownMetadataEntry | AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventPythonArtifactMetadataEntry | AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventFloatMetadataEntry | AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries_EventIntMetadataEntry;

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization {
  __typename: "Materialization";
  label: string;
  description: string | null;
  metadataEntries: AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization_metadataEntries[];
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_assetLineage_assetKey {
  __typename: "AssetKey";
  path: string[];
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_assetLineage {
  __typename: "AssetLineageInfo";
  assetKey: AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_assetLineage_assetKey;
  partitions: string[];
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent {
  __typename: "StepMaterializationEvent";
  runId: string;
  timestamp: string;
  stepKey: string | null;
  stepStats: AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_stepStats;
  materialization: AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_materialization;
  assetLineage: AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent_assetLineage[];
}

export interface AssetQuery_assetOrError_Asset_assetMaterializations {
  __typename: "AssetMaterialization";
  partition: string | null;
  runOrError: AssetQuery_assetOrError_Asset_assetMaterializations_runOrError;
  materializationEvent: AssetQuery_assetOrError_Asset_assetMaterializations_materializationEvent;
}

export interface AssetQuery_assetOrError_Asset {
  __typename: "Asset";
  key: AssetQuery_assetOrError_Asset_key;
  assetMaterializations: AssetQuery_assetOrError_Asset_assetMaterializations[];
}

export type AssetQuery_assetOrError = AssetQuery_assetOrError_AssetNotFoundError | AssetQuery_assetOrError_Asset;

export interface AssetQuery {
  assetOrError: AssetQuery_assetOrError;
}

export interface AssetQueryVariables {
  assetKey: AssetKeyInput;
}
