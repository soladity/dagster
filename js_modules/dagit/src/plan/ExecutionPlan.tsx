import * as React from "react";
import gql from "graphql-tag";
import styled from "styled-components/macro";
import { Colors } from "@blueprintjs/core";
import { ExecutionPlanFragment } from "./types/ExecutionPlanFragment";
import { ExecutionPlanBox } from "./ExecutionPlanBox";
import { AnimatedEllipsis } from "./AnimatedEllipsis";
import { ExecutionPlanHiddenStepsBox } from "./ExecutionPlanHiddenStepsBox";
import {
  IRunMetadataDict,
  IStepState,
  IStepMetadata
} from "../RunMetadataProvider";
import { RunFragment } from "../runs/types/RunFragment";
import { formatElapsedTime } from "../Util";

export interface IExecutionPlanProps {
  executionPlan: ExecutionPlanFragment;
  stepKeysToExecute?: (string | null)[] | null;
  runMetadata?: IRunMetadataDict;
  run?: RunFragment;
  onApplyStepFilter?: (step: string) => void;
  onShowStateDetails?: (step: string) => void;
  onReexecuteStep?: (step: string) => void;
}

const EMPTY_RUN_METADATA: IRunMetadataDict = {
  steps: {}
};

const EMPTY_STEP_METADATA: IStepMetadata = {
  state: IStepState.WAITING,
  start: undefined,
  elapsed: undefined,
  transitionedAt: 0,
  expectationResults: [],
  materializations: []
};

export class ExecutionPlan extends React.PureComponent<IExecutionPlanProps> {
  static fragments = {
    ExecutionPlanFragment: gql`
      fragment ExecutionPlanFragment on ExecutionPlan {
        steps {
          key
          kind
        }
        artifactsPersisted
      }
    `
  };

  renderSkippedSteps(skippedCount: number) {
    if (skippedCount > 0) {
      return <ExecutionPlanHiddenStepsBox numberStepsSkipped={skippedCount} />;
    }

    return null;
  }

  render() {
    const haveRun = !!this.props.runMetadata;
    const {
      onApplyStepFilter,
      onShowStateDetails,
      onReexecuteStep,
      runMetadata = EMPTY_RUN_METADATA,
      executionPlan,
      stepKeysToExecute,
      run
    } = this.props;

    const stepsOrderedByTransitionTime = Object.keys(runMetadata.steps).sort(
      (a, b) =>
        runMetadata.steps[a].transitionedAt -
        runMetadata.steps[b].transitionedAt
    );

    let startDone = false;
    let startText = null;

    if (haveRun) {
      startText = (
        <span>
          {`Loading`}
          <AnimatedEllipsis />
        </span>
      );
      if (runMetadata.startingProcessAt) {
        startText = (
          <span>
            {`Process starting`}
            <AnimatedEllipsis />
          </span>
        );
      }
      if (runMetadata.processId) {
        startText = (
          <span>
            {`Process (PID ${runMetadata.processId})`}
            <AnimatedEllipsis />
          </span>
        );
      }
      if (runMetadata.startedPipelineAt) {
        startDone = true;
        startText = (
          <span>
            {`Process (PID ${runMetadata.processId}) began pipeline execution`}
          </span>
        );
      } else if (runMetadata.initFailed) {
        startDone = true;
        startText = (
          <span>
            {`Process (PID ${runMetadata.processId}) pipeline initialization failed!`}
          </span>
        );
      }
      if (
        run &&
        run.pipeline &&
        run.pipeline.__typename === "UnknownPipeline"
      ) {
        startDone = true;
        startText = (
          <span>
            Could not load execution plan for pipeline {run.pipeline.name}.
          </span>
        );
      }
    }

    // Keep track of skipped steps that were not part of the current run
    let stepsSkippedCount = 0;
    let prevSkippedCount = 0;

    return (
      <ExecutionPlanContainer>
        <ExecutionPlanContainerInner>
          <ExecutionTimeline />
          <ExecutionTimelineMessage>
            <ExecutionTimelineDot completed={startDone} /> {startText}
          </ExecutionTimelineMessage>
          {executionPlan &&
            executionPlan.steps.map(step => {
              const delay =
                stepsOrderedByTransitionTime.indexOf(step.key) * 100;
              const metadata =
                runMetadata.steps[step.key] || EMPTY_STEP_METADATA;

              // If stepKeysToExecute is null, then all steps are being run
              const isStepPartOfRun =
                !stepKeysToExecute || stepKeysToExecute.indexOf(step.key) > -1;

              if (!isStepPartOfRun) {
                stepsSkippedCount++;
                return null;
              } else {
                prevSkippedCount = stepsSkippedCount;
                stepsSkippedCount = 0;
              }

              return (
                <div key={step.key}>
                  {this.renderSkippedSteps(prevSkippedCount)}
                  <ExecutionPlanBox
                    state={metadata.state}
                    start={metadata.start}
                    elapsed={metadata.elapsed}
                    key={step.key}
                    stepKey={step.key}
                    expectationResults={metadata.expectationResults}
                    materializations={metadata.materializations}
                    onShowStateDetails={onShowStateDetails}
                    onApplyStepFilter={onApplyStepFilter}
                    onReexecuteStep={onReexecuteStep}
                    executionArtifactsPersisted={
                      executionPlan.artifactsPersisted
                    }
                    delay={delay}
                  />
                </div>
              );
            })}
          {this.renderSkippedSteps(stepsSkippedCount)}
          {runMetadata.exitedAt && runMetadata.startedProcessAt && (
            <ExecutionTimelineMessage>
              <ExecutionTimelineDot completed={startDone} />
              <span>
                {`Process exited in ${formatElapsedTime(
                  runMetadata.exitedAt - runMetadata.startedProcessAt
                )}`}
              </span>
            </ExecutionTimelineMessage>
          )}
        </ExecutionPlanContainerInner>
      </ExecutionPlanContainer>
    );
  }
}

const ExecutionPlanContainer = styled.div`
  flex: 1;
  overflow-y: auto;
  color: ${Colors.WHITE};
  background: #232b2f;
`;

const ExecutionPlanContainerInner = styled.div`
  margin-top: 10px;
  position: relative;
  margin-bottom: 10px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  font-size: 0.9em;
`;

const ExecutionTimelineMessage = styled.div`
  display: flex;
  align-items: center;
  position: relative;
  color: ${Colors.LIGHT_GRAY2};
  margin-top: 3px;
  margin-bottom: 3px;
  z-index: 2;
`;

const ExecutionTimeline = styled.div`
  border-left: 1px solid ${Colors.GRAY3};
  position: absolute;
  top: 12px;
  left: 23px;
  bottom: 15px;
`;

const ExecutionTimelineDot = styled.div<{ completed: boolean }>`
  display: inline-block;
  width: 11px;
  height: 11px;
  border-radius: 5.5px;
  margin-right: 8px;
  background: #232b2f;
  border: 1px solid
    ${({ completed }) => (completed ? Colors.LIGHT_GRAY2 : Colors.GRAY1)};
  margin-left: 18px;
  flex-shrink: 0;
`;
