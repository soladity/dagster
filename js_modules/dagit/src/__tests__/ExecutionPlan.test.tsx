import * as React from "react";
import * as TestRenderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import { ExecutionPlan, IExecutionPlanProps } from "../plan/ExecutionPlan";
import { StepKind } from "../types/globalTypes";

it("renders given an execution plan", () => {
  const executionPlanProps: IExecutionPlanProps = {
    executionPlan: {
      __typename: "ExecutionPlan",
      artifactsPersisted: true,
      steps: [
        {
          __typename: "ExecutionStep",
          key: "raw_file_event_admins.compute",
          kind: StepKind.COMPUTE
        }
      ]
    }
  };
  const component = TestRenderer.create(
    <BrowserRouter>
      <ExecutionPlan {...executionPlanProps} />
    </BrowserRouter>
  );
  expect(component.toJSON()).toMatchSnapshot();
});
