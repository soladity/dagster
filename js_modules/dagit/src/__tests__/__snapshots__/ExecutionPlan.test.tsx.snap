// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders given an execution plan 1`] = `
<div
  className="ExecutionPlan__ExecutionPlanContainer-sc-1gu1caw-0 jvnsoS"
>
  <div
    className="ExecutionPlan__ExecutionPlanContainerInner-sc-1gu1caw-1 cIQpcx"
  >
    <div
      className="ExecutionPlan__ExecutionTimeline-sc-1gu1caw-3 jVhIJc"
    />
    <div
      className="ExecutionPlan__ExecutionTimelineMessage-sc-1gu1caw-2 fIocxs"
    >
      <div
        className="ExecutionPlan__ExecutionTimelineDot-sc-1gu1caw-4 idBHWu"
      />
       
    </div>
    <div>
      <div
        className="ExecutionPlanBox__ExecutionPlanRowContainer-sc-1bacjj9-5 kaLnAx"
      >
        <div
          className="ExecutionPlanBox__ExecutionPlanBoxContainer-sc-1bacjj9-6 bvnZYX waiting"
          onClick={[Function]}
          style={
            Object {
              "transitionDelay": "-100ms",
            }
          }
        >
          <div
            className="ExecutionPlanBox__ExecutionFinishedFlash-sc-1bacjj9-4 bGNKzx"
            style={
              Object {
                "transitionDelay": "-100ms",
              }
            }
          />
          <div
            style={
              Object {
                "alignItems": "center",
                "display": "inline-flex",
              }
            }
          >
            <div
              className="ExecutionPlanBox__ExecutionStateWrap-sc-1bacjj9-0 fipiZp"
              onClick={[Function]}
            >
              <div
                className="ExecutionStateDot-bprv14-0 fGCtEf"
                style={
                  Object {
                    "transitionDelay": "-100ms",
                  }
                }
                title="Waiting"
              />
              <div
                style={
                  Object {
                    "width": 4,
                  }
                }
              />
            </div>
            <div
              className="ExecutionPlanBox__ExecutionPlanBoxName-sc-1bacjj9-3 hFUwrq"
              title="raw_file_event_admins.compute"
            >
              raw_file_event_admins.compute
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
